File: backend-java/src/main/java/org/aiassistant/entities/Project.java
package org.aiassistant.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "projects")
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(length = 2000)
    private String description;

    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Document> documents = new ArrayList<>();

    private LocalDateTime createdDate;

    private LocalDateTime lastModifiedDate;

    @Column(name = "S3_Path")
    private String s3Path;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "json")
    private Map<String, String> fileSummaries;

    @PrePersist
    protected void onCreate() {
        this.createdDate = LocalDateTime.now();
        this.lastModifiedDate = LocalDateTime.now();
    }

    @PreUpdate
    protected void onModify() {
        this.lastModifiedDate = LocalDateTime.now();
    }
}

File: frontend-react/src/api/project.js
import axiosClient from "./axiosClient";

// GET /projects -> [{ id, name, description, createdDate, lastModifiedDate }]
export const getProjects = () => {
  return axiosClient.get("/projects");
};

// GET /projects/{id} -> { id, name, description, createdDate, lastModifiedDate }
export const getProjectById = (projectId) => {
  return axiosClient.get(`/projects/${projectId}`);
};

// POST /projects/save -> created project
export const createProject = (payload) => {
  return axiosClient.post("/projects/save", payload);
};
